### PR TITLE
TestCgroupHook use default max_attempts and updates to ncpus_are_core…

### PR DESCRIPTION
**Pull request by vccardenas**
…s test

#### Describe Bug or Feature
Use default max_attempts for the tests in TestCgroupHook and fix for intermitttent failure in test_cgroup_cpuset_ncpus_are_cores.

#### Describe Your Change
Removed instances of max_attempts in the tests except when existence=False is used in log_match().
Improved checking that the correct cgroups hook config match on the server and mom side.

#### Attach Test and Valgrind Logs/Output

[TestCgroupsHook.txt](https://github.com/openpbs/openpbs/files/5085737/TestCgroupsHook.txt)

Note: The failure in the test is unrelated to the issue and fix.

[TestCgroupsHook.test_cgroup_cpuset_ncpus_are_cores.txt](https://github.com/openpbs/openpbs/files/5093696/TestCgroupsHook.test_cgroup_cpuset_ncpus_are_cores.txt)

### CODE REVIEW

File: test/tests/functional/pbs_cgroups_hook.py [hunk 1]
```diff
@@ -3060,8 +3064,7 @@ def test_cgroup_periodic_update_known_jobs(self):
         self.tempfile.append(o)
         err_msg = "Unexpected error in pbs_cgroups " + \
             "handling exechost_periodic event: TypeError"
-        self.moms_list[0].log_match(err_msg, max_attempts=3,
-                                    interval=1, n=100,
+        self.moms_list[0].log_match(err_msg, interval=1, n=100,
```
Comment (lisa-altair) [inline test/tests/functional/pbs_cgroups_hook.py:3067]: Again, for existence=False leave the max_attempts alone.

File: test/tests/functional/pbs_cgroups_hook.py [hunk 2]
```diff
@@ -3147,8 +3150,7 @@ def test_cgroup_periodic_update_known_jobs(self):
...
```
Comment (lisa-altair) [inline test/tests/functional/pbs_cgroups_hook.py:3153]: leave this max_attempts in place.

File: test/tests/functional/pbs_cgroups_hook.py [hunk 3]
```diff
@@ -3159,7 +3161,7 @@ def test_cgroup_periodic_update_known_jobs(self):
         self.du.run_cmd(cmd=command, hosts=self.hosts_list[0], sudo=True)
         logmsg = '_exechost_periodic_handler: Failed to update %s' % jid1
         self.moms_list[0].log_match(msg=logmsg, starttime=presubmit,
-                                    max_attempts=1, existence=False)
+                                    existence=False)
```
Comment (lisa-altair) [inline test/tests/functional/pbs_cgroups_hook.py:3164]: leave this max_attempts in place

File: test/tests/functional/pbs_cgroups_hook.py [hunk 4]
```diff
@@ -3600,7 +3601,7 @@ def test_checkpoint_abort_preemption(self):
         self.server.expect(JOB, {'job_state': 'Q'}, id=jid1)
         err_msg = "%s;.*Failed to assign resources.*" % (jid2,)
         for m in self.moms.values():
-            m.log_match(err_msg, max_attempts=3, interval=1, n=100,
+            m.log_match(err_msg, interval=1, n=100,
```
Comment (lisa-altair) [inline test/tests/functional/pbs_cgroups_hook.py:3604]: leave this max_attempts in place.

File: test/tests/functional/pbs_cgroups_hook.py [hunk 5]
```diff
@@ -4136,7 +4137,7 @@ def test_cgroup_default_config(self):
         jid = self.server.submit(j)
         self.server.expect(JOB, {'job_state': 'R'}, jid)
         err_msg = "write_value: Permission denied.*%s.*memsw" % (jid)
-        self.mom.log_match(err_msg, max_attempts=3, interval=1, n=100,
+        self.mom.log_match(err_msg, interval=1, n=100,
```
Comment (lisa-altair) [inline test/tests/functional/pbs_cgroups_hook.py:4140]: leave this max_attempts in place

File: test/tests/functional/pbs_cgroups_hook.py [hunk 6]
```diff
@@ -1454,12 +1454,13 @@ def load_config(self, cfg, mom_checks=True):
         # are all finished, so that we don't match a CF copy
         # message in the logs from someone else!
         time.sleep(5)
+        just_before_import = time.time()
```
[the remark this answers is not in the document]
Comment (lisa-altair) [reply test/tests/functional/pbs_cgroups_hook.py:1457]: starttime can be a float

[the remark this answers is not in the document]
Comment (vstumpf) [reply test/tests/functional/pbs_cgroups_hook.py:1457]: oops, you're right!
just hte -1 should be ok then

**Comment by vccardenas** [comment]
@vstumpf, @bayucan please review once more.

**Pull request merged**